### PR TITLE
Sync internal code changes of tflite/tfhub.

### PR DESCRIPTION
**Pull request by mingxingtan**
Mostly contributed by Yiqi Li (yuqili@google.com)

### CODE REVIEW

File: efficientdet/dataloader_test.py [hunk 1]
```diff
@@ -13,63 +13,19 @@
...
 """Data loader and processing test cases."""
-import os
-import tempfile
+
 import tensorflow as tf
 
 import dataloader
 import hparams_config
-from dataset import tfrecord_util
+from brain_automl.efficientdet import test_util
```
Comment (fsx950223) [inline efficientdet/dataloader_test.py:21]: test_util

File: efficientdet/keras/inference_test.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 import tempfile
 from absl import logging
 import tensorflow as tf
+from brain_automl.efficientdet import test_util
```
Comment (fsx950223) [inline efficientdet/keras/inference_test.py:20]: import test_util

File: efficientdet/dataloader_test.py (hunk 1 above)
Comment (mingxingtan) [reply efficientdet/dataloader_test.py]: Good point. Done.

File: efficientdet/keras/inference_test.py (hunk 2 above)
Comment (mingxingtan) [reply efficientdet/keras/inference_test.py]: Done

**Comment by mingxingtan** [comment]
@ ziyeqinghan fyi

**Comment by fsx950223** [comment]
Add tensorflow_hub dependency.

File: efficientdet/keras/train_lib.py [hunk 3]
```diff
@@ -843,3 +849,61 @@ def test_step(self, data):
...
+    # class/box output prediction network.
+    num_anchors = len(config.aspect_ratios) * config.num_scales
+
+    if config.separable_conv:
+      conv2d_layer = functools.partial(
+          tf.keras.layers.SeparableConv2D, depth_multiplier=1)
+    else:
+      conv2d_layer = tf.keras.layers.Conv2D
+    self.classes = conv2d_layer(
+        config.num_classes * num_anchors,
```
Comment (fsx950223) [inline efficientdet/keras/train_lib.py:872]: Initializers are different from original one.

**Review by fsx950223 — APPROVED** [review]

File: efficientdet/keras/train_lib.py (hunk 3 above)
Comment (mingxingtan) [reply efficientdet/keras/train_lib.py:872]: It seems to be the same as before: https://github.com/google/automl/blob/890eaddc0963701e515de06a11a662d1f1027770/efficientdet/keras/efficientdet_keras.py#L426

Notably, the last convnet for class prediction is special in order to enable focal loss.

Comment (fsx950223) [reply efficientdet/keras/train_lib.py:872]: ```pointwise_initializer``` and ```depthwise_initializer``` are ```tf.initializers.variance_scaling()``` in ClassNet, should it be changed? It's better to use ClassNet and BoxNet whose are defined in efficientnet_keras.py

Comment (mingxingtan) [reply efficientdet/keras/train_lib.py:872]: Good point. I am going to merge this for now, and update it later.

**Review by fsx950223 — APPROVED** [review]
